app/src/main/java/org/wikipedia/dataclient/RestService.kt

```diff
@@ -0,0 +1,210 @@
+import org.wikipedia.suggestededits.provider.SuggestedEditItem
+import retrofit2.Call
+import retrofit2.Response
+import retrofit2.http.*
+
+interface RestService {
+
+    /**
+     * Gets a page summary for a given title -- for link previews
+     *
+     * @param title the page title to be used including prefix
+     */
+    @Headers("x-analytics: preview=1", "Accept: $ACCEPT_HEADER_SUMMARY")
+    @GET("page/summary/{title}")
+    fun getSummaryResponse(
+        @Path("title") title: String,
+        @Header("Referer") referrerUrl: String?,
+        @Header("Cache-Control") cacheControl: String?,
+        @Header(OfflineCacheInterceptor.SAVE_HEADER) saveHeader: String?,
+        @Header(OfflineCacheInterceptor.LANG_HEADER) langHeader: String?,
+        @Header(OfflineCacheInterceptor.TITLE_HEADER) titleHeader: String?
+    ): Observable<Response<PageSummary>>
+
+    @Headers("x-analytics: preview=1", "Accept: $ACCEPT_HEADER_SUMMARY")
+    @GET("page/summary/{title}")
+    fun getSummary(
+        @Header("Referer") referrerUrl: String?,
+        @Path("title") title: String
+    ): Observable<PageSummary>
+
+    // todo: this Content Service-only endpoint is under page/ but that implementation detail should
+    //       probably not be reflected here. Move to WordDefinitionClient
+    /**
+     * Gets selected Wiktionary content for a given title derived from user-selected text
+     *
+     * @param title the Wiktionary page title derived from user-selected Wikipedia article text
+     */
+    @Headers("Accept: $ACCEPT_HEADER_DEFINITION")
+    @GET("page/definition/{title}")
+    fun getDefinition(@Path("title") title: String): Observable<Map<String, Array<RbDefinition.Usage>>>
+
+    @get:GET("page/random/summary")
+    @get:Headers("Accept: $ACCEPT_HEADER_SUMMARY")
+    val randomSummary: Observable<PageSummary>
+
+    @Headers("Accept: $ACCEPT_HEADER_SUMMARY")
+    @GET("page/related/{title}")
+    fun getRelatedPages(@Path("title") title: String?): Observable<RbRelatedPages>
+
+    @GET("page/media-list/{title}/{revision}")
+    fun getMediaList(
+        @Path("title") title: String,
+        @Path("revision") revision: Long
+    ): Observable<MediaList>
+
+    @GET("page/media-list/{title}/{revision}")
+    fun getMediaListResponse(
+        @Path("title") title: String?,
+        @Path("revision") revision: Long,
+        @Header("Cache-Control") cacheControl: String?,
+        @Header(OfflineCacheInterceptor.SAVE_HEADER) saveHeader: String?,
+        @Header(OfflineCacheInterceptor.LANG_HEADER) langHeader: String?,
+        @Header(OfflineCacheInterceptor.TITLE_HEADER) titleHeader: String?
+    ): Observable<Response<MediaList>>
+
+    @GET("feed/onthisday/events/{mm}/{dd}")
+    fun getOnThisDay(@Path("mm") month: Int, @Path("dd") day: Int): Observable<OnThisDay>
+
+    @get:GET("feed/announcements")
+    @get:Headers("Accept: " + ACCEPT_HEADER_PREFIX + "announcements/0.1.0\"")
```

Comment:
Not sure but do you agree that we should use either `@GET()` and `@Headers` with a `fun` style method instead of `val` style in the `Service` related classes? Somehow the `@get:Get()` looks a bit messy to me.

Reply:
The reason I left them this way is because if we change them to functions, we would also need to change all of the places they are used to function invocation`()` style, which would require modifying 100 files.

Can we leave them like this, and then update them one-by-one as we transition them to be `suspend` functions?

Reply:
I see. It sounds good to me!